src/coreclr/interpreter/compiler.cpp

```diff
@@ -21,6 +21,7 @@ static const StackType g_stackTypeFromInterpType[] =
     StackTypeI4, // I2
     StackTypeI4, // U2
     StackTypeI4, // I4
+    StackTypeI4, // U4
```

Comment:
Do we really need this ? `LDELEM_U4` is identical to `LDELEM_I4`

Reply:
Not needed. Removed it and updated other LDELEM cases for types smaller than 32 bits to use int32_t as the target type.